src/App.tsx

```diff
@@ -1,223 +1,243 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+
+import {
+  createTodo,
+  deleteTodoById,
+  getTodos,
+  updateTodoById,
+} from './api/todos';
+
+import { filterTodos } from './helpers/completedFilter';
+
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [complitedFilter, setComplitedFilter] = useState(FilterType.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isNewTodoLoading, setIsNewTodoLoading] = useState(false);
+  const [updatingTodos, setUpdatingTodos] = useState<number[]>([]);
+
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => {
+          setErrorMessage('Unable to download todos');
+        });
+    }
+  }, []);
+
+  const uncompletedTodosLength = useMemo(() => (
+    todos.filter(todo => !todo.completed).length
+  ), [todos]);
+
+  const completedTodosLength = useMemo(() => (
+    todos.filter(todo => todo.completed).length
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return filterTodos(todos, complitedFilter);
+  }, [complitedFilter, todos]);
+
+  const deleteTodo = useCallback(async (todoId: number) => {
+    try {
+      const deleteResponse = await deleteTodoById(todoId);
+
+      setTodos(prevTodos => (
+        prevTodos.filter(todo => todo.id !== todoId)
+      ));
+
+      return deleteResponse;
+    } catch (deleteError) {
+      return setErrorMessage('Unable to delete a todo');
+    }
   }, []);
 
+  const clearCompletedTodos = useCallback(() => {
+    todos.forEach(todo => {
+      if (todo.completed) {
+        deleteTodo(todo.id);
+      }
+    });
+  }, [todos]);
+
+  const addTodo = useCallback(
+    async (event: React.FormEvent<HTMLFormElement>, title: string) => {
+      event.preventDefault();
```

Comment:
What if you do this inside `handleAddTodo` function ? So you don't need to pass the event as param to `addTodo` function

Reply:
fixed